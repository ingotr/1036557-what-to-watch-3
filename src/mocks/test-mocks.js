const FirstMovie = {
  TITLE: `Mock-test-The Grand Budapest Hotel`,
  GENRE: `Mock-test-Drama`,
  RELEASE_DATE: 2020,
};

const movies = [
  {
    id: `mock-test-011`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-012`,
    title: `Bohemian Rhapsody`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-013`,
    title: `Macbeth`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-014`,
    title: `Aviator`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-015`,
    title: `We need to talk about Kevin`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-016`,
    title: `What We Do in the Shadows`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-017`,
    title: `Revenant`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-018`,
    title: `Johnny English`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];

const movieInfo = {
  title: `Mock-test-The Grand Budapest Hotel`,
  genre: `Mock-test-Drama`,
  year: 2017,
  poster: {
    big: `img/the-grand-budapest-hotel-poster.jpg`,
    bigAlt: `Mock-test-The Grand Budapest Hotel poster`,
  },
  rating: {
    score: 8.9,
    level: `Mocks-Very good`,
    count: 240,
  },
  director: `Mock-Wes Andreson`,
  description: [`Mock-test-In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave &#96s friend and protege.`,
    `Gustave prides himself on providing first-className service to the hotel &#96s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave &#96s lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
  starring: [`Mock-test-Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
};

export {FirstMovie, movies, movieInfo};
