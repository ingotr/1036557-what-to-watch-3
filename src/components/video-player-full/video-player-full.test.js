import React from 'react';
import renderer from 'react-test-renderer';
import VideoPlayerFull from './video-player-full.jsx';
import {film} from '../../mocks/test-mocks.js';

it(`<FullVideoPlayer /> should render correctly`, () => {
  const tree = renderer
    .create(
        <VideoPlayerFull
          film={film}
          onItemLeave={() => {}}
        />, {
          createNodeMock: () => {
            return {};
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
