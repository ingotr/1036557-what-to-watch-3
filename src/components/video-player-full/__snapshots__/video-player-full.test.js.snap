// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullVideoPlayer /> should render correctly 1`] = `
<div
  className="player"
  style={
    Object {
      "zIndex": 10,
    }
  }
>
  <video
    autoPlay={true}
    className="player__video"
    poster={
      Object {
        "big": "img/the-grand-budapest-hotel-poster.jpg",
        "bigAlt": "Mock-test The Grand Budapest Hotel poster",
      }
    }
  />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value={0}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        0:00:0
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="21"
          viewBox="0 0 14 21"
          width="14"
        >
          <use
            xlinkHref="#pause"
          />
        </svg>
        <span>
          Pause
        </span>
      </button>
      <div
        className="player__name"
      >
        Mock-test The Grand Budapest Hotel
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
